### PR TITLE
Add more functions

### PR DESCRIPTION
**Pull request by codewithgun**
- imbalance deposit
- withdraw single side
- depeg pool

### CODE REVIEW

File: ts-client/src/constants.ts [hunk 1]
```diff
@@ -10,3 +10,8 @@ export const ERROR = {
   POOL_NOT_LOAD: "Pool not loaded",
   INVALID_MINT: "Invalid mint",
 };
+
+export const EXTRA_ACCOUNTS = {
+  marinade: [new PublicKey("8szGkuLTAux9XMgZ2vtY39jVSowEcpBfFfD8hXSEqdGC")],
```
Comment (andrewsource147) [inline ts-client/src/constants.ts:15]: Could you comment how the account is used for

File: ts-client/src/curve/stable-swap.ts [hunk 2]
```diff
@@ -1,70 +1,385 @@
...
     this.amp = amp;
+    this.tokenMultiplier = tokenMultiplier;
+    this.depeg = depeg;
+    this.onChainTime = onChainTime;
+    this.extraAccounts = extraAccounts;
+  }
+
+  private getBasePoolVirtualPrice(depegType: DepegType): BN {
+    if (depegType["marinade"]) {
+      const account = this.extraAccounts.get(
```
Comment (andrewsource147) [inline ts-client/src/curve/stable-swap.ts:51]: Should we add check whether the account is not null?

File: ts-client/src/curve/stable-swap.ts [hunk 3]
```diff
@@ -1,70 +1,385 @@
...
+    const { tokenBMultiplier } = this.tokenMultiplier;
+    return tokenBAmount.div(tokenBMultiplier);
+  }
+
+  private upscaleTokenA(tokenAAmount: BN): BN {
+    const normalizedTokenAAmount = this.normalizeTokenA(tokenAAmount);
+    return this.pegTokenA(normalizedTokenAAmount);
+  }
+
+  private downscaleTokenA(tokenAAmount: BN): BN {
```
Comment (andrewsource147) [inline ts-client/src/curve/stable-swap.ts:138]: I think we can wrap all logic: for normalize + pegPrice in 1 function

File: ts-client/src/constants.ts (hunk 1 above)
Comment (codewithgun) [reply ts-client/src/constants.ts:17]: Added comment

File: ts-client/src/curve/stable-swap.ts (hunk 3 above)
Comment (codewithgun) [reply ts-client/src/curve/stable-swap.ts]: Fixed

**Pull request merged**